src/seismometer/controls/explore.py

```diff
@@ -0,0 +1,907 @@
+            if scores should be grouped by context, by default True
+        fairness_metrics : tuple[str], optional
+            list of fairness metrics to display, if None (default) will use ["tpr", "fpr", "pprev"]
+        fairness_threshold : float, optional
+            threshold for fairness metrics, by default 1.25
+        """
+        all_metrics = ["pprev", "tpr", "fpr", "fnr", "ppr", "precision"]
+        fairness_metrics = fairness_metrics or ["pprev", "tpr", "fpr"]
+        self.fairness_slider = FloatSlider(
+            description="Audit Threshold",
+            value=fairness_threshold,
+            min=1.0,
+            max=2.0,
+            step=0.01,
+            tooltip="Threshold for fairness metrics",
+            style=WIDE_LABEL_STYLE,
+        )
+        thresholds = {"Score Threshold": score_threshold}
+        self.fairness_list = SelectionListWidget(options=all_metrics, value=fairness_metrics, title="Audit Metrics")
+        self.model_options = ModelOptionsWidget(target_names, score_names, thresholds, per_context)
+        self.fairness_list.observe(self._on_value_change, "value")
+        self.fairness_slider.observe(self._on_value_change, "value")
+        self.model_options.observe(self._on_value_change, "value")
+
+        super().__init__(
+            children=[self.model_options, self.fairness_list, self.fairness_slider], layout=BOX_GRID_LAYOUT
+        )
+        self._disabled = False
+
+    @property
+    def disabled(self) -> bool:
+        return self._disabled
+
+    @disabled.setter
+    def disabled(self, disabled: bool):
+        self._disabled = disabled
+        self.fairness_list.disabled = disabled
+        self.fairness_slider.disabled = disabled
+        self.model_options.disabled = disabled
+
+    def _on_value_change(self, change=None):
+        self.value = {
+            "fairness_metrics": self.fairness_list.value,
+            "fairness_threshold": self.fairness_slider.value,
+            "model_options": self.model_options.value,
+        }
+
+    @property
+    def metrics(self) -> tuple[str]:
+        """selected cohorts"""
+        return self.fairness_list.value
+
+    @property
+    def fairness_threshold(self) -> tuple[str]:
+        """selected cohorts"""
+        return self.fairness_slider.value
+
+    @property
+    def target(self) -> str:
+        """trarget column descriptor"""
+        return self.model_options.target
+
+    @property
+    def score(self) -> str:
+        """Score column descriptor"""
+        return self.model_options.score
+
+    @property
+    def score_threshold(self) -> tuple[float]:
+        """Score thresholds"""
+        return self.model_options.thresholds["Score Threshold"]
+
+    @property
+    def group_scores(self) -> bool:
+        """If scores should be grouped by context"""
+        return self.model_options.group_scores
+
+
+# endregion
+# region Exploration Widgets (Intermediate Classes)
+
+
+class ExplorationWidget(VBox):
```

Comment:
Nit: is this really "Intermediate"?  It's still based on VBox, and the other ones in this region are mostly based on this one

Reply:
FINC